src/server/search/search_family.cc

```diff
@@ -110,6 +117,47 @@ optional<SearchParams> ParseSearchParamsOrReply(CmdArgList args, ConnectionConte
       continue;
     }
 
+    // RETURN {num} [{ident} AS {name}...]
+    if (ArgS(args, i) == "RETURN") {
+      if (i + 1 >= args.size()) {
+        (*cntx)->SendError(kSyntaxErr);
+        return nullopt;
+      }
+
+      uint64_t num_fields = args.size();
+      if (!absl::SimpleAtoi(ArgS(args, i + 1), &num_fields)) {
```

Comment:
Something that popped in my mind - and this kinda applies for all of the commands as well. Each element in the `MutableSlice` is a `char` which means that the largest integer it can represent is 255 (assuming unsigned). What if `limit` > `255`?

Reply:
No they're parsed as as string 🤣 CmdArgList is absl::Span<MutableSlice> so its a string_view[]

Reply:
ohhh yes yes I brainfarted, so the whole string_view is the number. Lol